### PR TITLE
fix for deleted fields filters in segments

### PR DESCRIPTION
**Pull request by mqueme**
[//]: # ( Please answer the following questions: )

| Q  | A
| --- | ---
| Bug fix? | y
| New feature? | 
| Related user documentation PR URL | 
| Related developer documentation PR URL | 
| Issues addressed (#s or URLs) | #2582
| BC breaks? | 
| Deprecations? | 

[//]: # ( Note that all new features should have a related user and/or developer documentation PR in their respective repositories. )

[//]: # ( Required: )
#### Description:
This PR fixes bug when a custom field that is used in a segment filter is deleted
[//]: # ( As applicable: )
#### Steps to reproduce the bug:
1. create a custom field
2. use it in a segment filter and update the segment
3. there should be an error

#### Steps to test this PR:
1. same as above, but the error is gone when updating segments

### CODE REVIEW

File: app/bundles/LeadBundle/Entity/LeadListRepository.php [hunk 1]
```diff
@@ -1113,7 +1116,7 @@ public function getListFilterExpr($filters, &$parameters, QueryBuilder $q, $not
                             $groupExpr->add(
                                 $this->generateFilterExpression($q, $field, $func, $details['filter'], null)
                             );
-
+                            echo 'hola';
```
Comment (escopecz) [inline app/bundles/LeadBundle/Entity/LeadListRepository.php]: 😉

**Comment by escopecz** [comment]
I'm getting this waring on `app/console mautic:segments:update` command:
`Undefined variable: expr in app/bundles/LeadBundle/Entity/LeadListRepository.php on line 1193`

**Comment by escopecz** [comment]
It doesn't throw any error now, but I noticed some odd behaviour:
1. Create a custom field
2. Use this custom field in a random segment filter.
3. Run command to update segments - The segment has no contacts assigned to it, that's correct since the filter was not matching any contacts.
4. Remove the custom field created in 1.
5. Run command to update segments - All contacts are assigned to this contact. Why?

**Comment by mqueme** [comment]
it is no longer adding all fields to lists with deleted custom fields

**Comment by escopecz** [comment]
I got this error this time:
```
Rebuilding contacts for segment 17
PHP Notice:  Undefined variable: expr in /app/bundles/LeadBundle/Entity/LeadListRepository.php on line 321
PHP Stack trace:
PHP   1. {main}() /app/console:0
PHP   2. Symfony\Component\Console\Application->run() /app/console:41
PHP   3. Symfony\Bundle\FrameworkBundle\Console\Application->doRun() /vendor/symfony/console/Application.php:124
PHP   4. Symfony\Component\Console\Application->doRun() /vendor/symfony/framework-bundle/Console/Application.php:92
PHP   5. Symfony\Component\Console\Application->doRunCommand() /vendor/symfony/console/Application.php:193
PHP   6. Symfony\Component\Console\Command\Command->run() /vendor/symfony/console/Application.php:872
PHP   7. Mautic\LeadBundle\Command\UpdateLeadListsCommand->execute() /vendor/symfony/console/Command/Command.php:257
PHP   8. Mautic\LeadBundle\Model\ListModel->rebuildListLeads() /app/bundles/LeadBundle/Command/UpdateLeadListsCommand.php:80
PHP   9. Mautic\LeadBundle\Model\ListModel->getLeadsByList() /app/bundles/LeadBundle/Model/ListModel.php:517
PHP  10. Mautic\LeadBundle\Entity\LeadListRepository->getLeadsByList() /app/bundles/LeadBundle/Model/ListModel.php:988

  [Symfony\Component\Debug\Exception\FatalThrowableError]
  Call to a member function count() on null
```

**Comment by dongilbert** [comment]
This worked as expected, I did not encounter the error pointed out by @escopecz

**Comment by escopecz** [comment]
This is segment 17 if you'd like to try replicate
screen shot 2017-04-18 at 21 10 57
:

**Pull request merged**